src/colorizer/loaders/UrlArrayLoader.ts

```diff
@@ -59,11 +59,6 @@ export default class UrlArrayLoader implements IArrayLoader {
    * @returns a URLArraySource object containing the loaded data.
    */
   async load<T extends FeatureDataType>(url: string, type: T, min?: number, max?: number): Promise<UrlArraySource<T>> {
-    if (!url.endsWith(".json") && !url.endsWith(".parquet")) {
-      throw new Error(
-        `Encountered unsupported file format when loading data. URL must end in '.parquet' or '.json': ${url}`
-      );
-    }
```

Comment:
This check now happens in the worker. I removed the strict requirement for JSON or Parquet file extensions, because file URLs (as generated by `createObjectURL`) don't have file extension info associated with them.